Tail.scala

```diff
@@ -0,0 +1,226 @@
+// Author: James Dunn, UC Berkeley (dunn@eecs.berkeley.edu)
```

Comment:
I would remove all author contributions in the code and instead add it to the README.